rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPoolPartition.java

```diff
@@ -63,6 +66,7 @@ public ObjectPoolPartition(ObjectPool<T> pool, PoolConfig config,
 
   public void returnObject(Poolable<T> object)
   {
+    takeSemaphore.release();
```

Comment:
remove `takeSemaphore.release()` here, move rest of the code in try-finally and call `takeSemaphore.release()` in the finally block. Otherwise current code opens up the case of going beyond max connections

Reply:
makes sense, fixed.